homeassistant/components/tts/__init__.py

```diff
@@ -910,11 +924,15 @@ def async_remove_from_mem(_: datetime) -> None:
             ),
         )
 
-    async def async_read_tts(self, filename: str) -> tuple[str | None, bytes]:
+    async def async_read_tts(self, token: str) -> tuple[str | None, bytes]:
         """Read a voice file and return binary.
 
         This method is a coroutine.
         """
+        filename = self.filename_token.get(token)
+        if not filename:
+            raise HomeAssistantError(f"{token} was not recognized!")
+
         if not (record := _RE_VOICE_FILE.match(filename.lower())) and not (
             record := _RE_LEGACY_VOICE_FILE.match(filename.lower())
```

Comment:
This is no longer needed now

Reply:
It's still needed to reconstruct the cache key correctly right below. I'll save this clean up for a future PR.